packages/material-ui/src/ButtonGroup/ButtonGroup.test.js

```diff
@@ -152,7 +152,7 @@ describe('<ButtonGroup />', () => {
         <Button>Hello World</Button>
       </ButtonGroup>,
     );
-    assert.strictEqual(wrapper.childAt(0).props().size, 'large');
+    assert.strictEqual(wrapper.find('button').hasClass(classes.sizeLarge), true);
```

Comment:
That test fails (find doesn't find anything), but also wouldn't the ButtonBase test only be concerned that the prop has been passed? It's up to the Button tests to check that the correct class is applied, otherwise we're testing Button from within ButtonBase tests.

Reply:
Yes these hybrid unit tests turned out to cost very much during refactoring. Switching to a11y selectors is more robust and actually test what the component user sees.

Reply:
Wrong convo but the principle applies. Nobody actually cares how the large prop is passed along through the component tree. Expecting a css class is still an implementation detail but it's far less likely to break during refactoring.